resources/assets/js/components/teams/TeamMembershipButton.vue

```diff
@@ -0,0 +1,89 @@
+<template>
+    <div>
+        <button
+                type="button" class="btn btn-secondary"
+                :disabled="!this.user.is_active || !this.team.self_serviceable"
+                v-if="memberOfTeam" v-on:click="changeMembership('leave')">
+            Leave
+        </button>
+        <button
+                type="button" class="btn btn-primary"
+                :disabled="!this.user.is_active || !this.team.self_serviceable"
+                v-else v-on:click="changeMembership('join')">
+            Join
+        </button>
+        <em v-if="!this.user.is_active">&nbsp;&nbsp;<small>You must be an active member to join/leave teams.</small></em>
+        <em v-if="!this.team.self_serviceable">&nbsp;&nbsp;<small>Join/leave is restricted for this team.</small></em>
+    </div>
+</template>
+<script>
+export default {
+  props: {
+    team: {},
+    user: {},
+  },
+  data() {
+    return {
+      baseUrl: '/api/v1/teams/',
+      user_teams: this.user.teams,
+    };
+  },
+  computed: {
+    memberOfTeam: function() {
+      let self = this;
+      return this.user_teams.some(function(val) {
+        return val.id === self.team.id;
+      });
+    },
+  },
+  methods: {
+    changeMembership: function(action) {
+      let data = {
+        user_id: this.user.id,
+        team_id: this.team.id,
+        action: action,
+      };
+      axios
+        .post(this.baseUrl + this.team.id + '/members', data)
+        .then(response => {
+          if (response.status === 201 && data.action === 'join') {
+            swal({
+              type: 'success',
+              title: 'Sweet!',
+              text: 'You joined ' + this.team.name + '!',
```

Comment:
I don't like the verbiage here but I don't know what I want it to be yet.

Reply:
Can you add a note somewhere somehow that people should join the slack and mailing list for the team?

I want it to be clear that joining a group doesn't do anything useful (outside of SSO which I'm going to continue to handwave as magic.)